test/scale/run_test.py

```diff
@@ -0,0 +1,372 @@
+                
+                for idx, char in enumerate(line):            
+                    if status == 0:
+                        if char == '$':
+                            status = 1
+                        else:
+                            output_line_parts.append(char)
+                    elif status == 1:
+                        if char == '{':
+                            status = 2
+                        else:
+                            output_line_parts.append(f"${char}")
+                            status = 0
+                    elif status == 2:
+                        if not char.isspace():
+                            parameter_start = idx
+                            status = 3
+                    elif status == 3:
+                        if char.isspace():
+                            parameter_end = idx
+                            status = 4
+                        elif char == '}':
+                            parameter_end = idx
+                            output_line_parts.append(
+                                self.configurations[
+                                    line[parameter_start:parameter_end]
+                                ]
+                            )
+                            parameter_start = -1
+                            status = 0
+                    elif status == 4:
+                        if char == '}':
+                            output_line_parts.append(
+                                self.configurations[
+                                    line[parameter_start:parameter_end]
+                                ]
+                            )
+                            parameter_start = -1
+                            status = 0
+                
+                if status != 0:
+                    raise Exception(
+                        f"template in '{line}' doesn't end in one line"
+                    )
+
+                target_file.write("".join(output_line_parts))
+        
+        logger.info("generated workload yaml:")
+        self.run_command(f"cat {self._generated_workload_yaml}")
+    
+    def run_command(self, command: str, need_stdout=False, raise_error=True):
+        logger.info(f"run command {command}")
+        stdout = None
+        if need_stdout:
+            stdout = subprocess.PIPE
+        process = subprocess.run(command, shell=True, text=True, stdout=stdout)
+        if process.returncode != 0 and raise_error:
+            raise RuntimeError(f"command {command} exit with error"
+                               f" code {process.returncode}")
+        stdout = process.stdout
+        if need_stdout:
+            logger.info(stdout)
+        return stdout
+    
+    def setup(self, current_scale):
+        logger.info("reinstalling CSI driver")
+        self.run_command(f"{ROOT_PATH}/deploy/uninstall-driver.sh")
+        self.run_command(f"{ROOT_PATH}/deploy/install-driver.sh")
+        os.mkdir(self._csi_log_path)
+
```

Comment:
Check CSI driver State:

kubectl wait pod -n kube-system --for=condition=Ready --selector='app in (csi-azurelustre-controller,csi-azurelustre-node)' --timeout=300s

Reply:
Hi @vinli-cn. I encountered an problem. `kubectl wait` keeps waiting after all pods in node daemonset is ready. Maybe some of wait's mechanisms don't work well with deamonset together? I'm still trying some new way to check the driver is ready.

Reply:
I think we can add the `--wait` flag in CSI installation script.